### PR TITLE
Enhanced user experience by integrating a color picker

### LINKED ISSUE #4227: ⭐️ Feature : Add Color Picker for border color
### Is your feature request related to a problem? Please describe

It is hard to first pick a good color and place it in input box.

### Describe the solution you'd like

Easier option is to provide a color picker to navigate between colors and choose a great color.
### Video of Solution :
https://github.com/CircuitVerse/CircuitVerse/assets/77102885/15e1fadc-7b85-4b2d-acf4-dc3083427bee

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by shauryag2002**
Fixes #4227 

#### Describe the changes you have made in this PR -
Simplified user interaction by implementing a color picker for seamless navigation and effortless selection of vibrant colors.
### Video of the Feature -

https://github.com/CircuitVerse/CircuitVerse/assets/77102885/213a2b9b-3b05-4bda-80f7-83a338a2dc87

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: app/component/project_embed_component.html.erb [hunk 1]
```diff
@@ -21,6 +21,7 @@
         <div class="field form-group embed-input-fields">
             <div><label for="border_px" class="col-sm-2 col-form-label"><b><%= t("projects.modal.border") %></b></label></div>
             <input type="text" class="form-control form-input" id="border_px" placeholder="Width (e.g. 2px)">&nbsp;&nbsp;
+            <input type="color" class="w-25 h-auto" id="color_picker">
```
[the remark this answers is not in the document]
Comment (shauryag2002) [reply app/component/project_embed_component.html.erb]: @tanmoysrt Can you tell me , What I exactly need to add for this accesslint (bot) issue ?

**Review by tanmoysrt — CHANGES_REQUESTED** [review]
Hi @shauryag2002 
The color picker is working fine.

There is an issue.
Suppose user  has updated colour hex code from the input box, then you should also sync with the colour picker so that always the colour representing by color picker and hex code in textbox is same

File: app/component/project_embed_component.html.erb [hunk 2]
```diff
@@ -96,3 +97,13 @@
...
 </div>
+<script>
+  const color_picker = document.getElementById('color_picker');
+  const colorDisplay = document.getElementById('border_color');
+  color_picker.addEventListener('change', function() {
+      colorDisplay.value = color_picker.value;
+  });
+  colorDisplay.addEventListener('blur',function(){
+      color_picker.value = colorDisplay.value;
+  });
```
Comment (vedant-jain03) [inline app/component/project_embed_component.html.erb:108]:
```suggestion
  color_picker.addEventListener('input', function() {
      colorDisplay.value = color_picker.value;
  });
  document.getElementById('border_color').addEventListener('input', (e) => {
    color_picker.value = e.target.value
  })
```

Thanks for this. LGTM
We can make it dynamic through `input` event listener

**Review by vedant-jain03 — CHANGES_REQUESTED** [review]

**Review by vedant-jain03 — APPROVED** [review]
Thanks for the changes @shauryag2002 🚀

**Review by tanmoysrt — APPROVED** [review]
Looks good !
Thanks @shauryag2002

**Pull request merged**